app/ante/ante.go

```diff
@@ -177,9 +182,35 @@ func IsSystemTx(tx sdk.Tx, isAuthorizedSigner func(string) bool) bool {
 		*observertypes.MsgAddBlameVote:
 		signers := innerMsg.GetSigners()
 		if len(signers) == 1 {
-			return isAuthorizedSigner(signers[0].String())
+			if isObserverAuthorized(signers[0].String()) {
+				return true
+			}
 		}
 	}
 
 	return false
 }
+
+// IsAuthorityTx checks if there is a MsgCreateValidator in the transaction and returns the signer if present.
+func IsAuthorityTx(tx sdk.Tx) (bool, string) {
+	for _, msg := range tx.GetMsgs() {
+		switch msg.(type) {
+		case *stakingtypes.MsgCreateValidator: // Other message types can be added here
+			signers := msg.GetSigners()
+			if len(signers) >= 1 {
+				return true, signers[0].String()
+			}
+		}
+	}
+	return false, "" // No MsgCreateValidator found, return false and an empty string
+}
+
+func handleAuthorityTx(ctx sdk.Context, msgCreateSigner string, options HandlerOptions) (sdk.AnteHandler, error) {
+	if options.AuthorityKeeper.IsAuthorized(ctx, msgCreateSigner, authoritytypes.PolicyType_groupAdmin) {
+		return newCosmosAnteHandler(options), nil
+	} else if ctx.BlockHeight() == 0 {
+		return newCosmosAnteHandlerForSystemTx(options), nil
+	} else {
+		return nil, errorsmod.Wrapf(errortypes.ErrUnauthorized, "not authorized for AuthorityTx: %s", msgCreateSigner)
+	}
+}
```

Comment:
This nested if statement will lead bug, how about this swihtch/case
```suggestion
func handleAuthorityTx(ctx sdk.Context, msgCreateSigner string, options HandlerOptions) (sdk.AnteHandler, error) {
	switch {
	case options.AuthorityKeeper.IsAuthorized(ctx, msgCreateSigner, authoritytypes.PolicyType_groupAdmin):
		return newCosmosAnteHandler(options), nil
	case ctx.BlockHeight() == 0:
		return newCosmosAnteHandlerForSystemTx(options), nil
	default:
		return nil, errorsmod.Wrapf(errortypes.ErrUnauthorized, "not authorized for AuthorityTx: %s", msgCreateSigner)
	}
}
```